libs/gl-client/src/pairing/attestation_device.rs

```diff
@@ -148,4 +150,119 @@ impl Client<Connected> {
             .await?
             .into_inner())
     }
+
+    pub fn verify_pairing_data(data: GetPairingDataResponse) -> Result<()> {
+        let mut crs = std::io::Cursor::new(&data.csr);
+        let pem = Pem::read_from(&mut crs).map_err(into_verify_pairing_data_error)?;
+        let csr = Csr::from_pem(&pem).map_err(into_verify_pairing_data_error)?;
+        let sub_pk_der = csr
+            .public_key()
+            .to_der()
+            .map_err(into_verify_pairing_data_error)?;
+        let sub_pk_info: SubjectPublicKeyInfo =
+            picky_asn1_der::from_bytes(&sub_pk_der).map_err(into_verify_pairing_data_error)?;
+
+        if let PublicKey::Ec(bs) = sub_pk_info.subject_public_key {
+            let pk = hex::encode(bs.0.payload_view());
+
+            if pk == data.session_id
+                && Self::restriction_contains_pubkey_exactly_once(&data.restrs, &data.session_id)
+            {
+                Ok(())
+            } else {
+                Err(Error::VerifyPairingDataError(format!(
+                    "public key {} does not match pk {}",
+                    data.session_id, pk
+                )))
```

Comment:
This is a bit of circular message, can we come up with a more clear way of phrasing this? "The public key you asked us to sign does not match the session_id"?